### DISCUSSION TITLE
Page navigation buttons - how to configure?

### DISCUSSION THREAD

**Discussion by nagyrobi** [HASwitchPlate/openHASP#78] [Q&A] [answered]
Trying the [examples](https://fvanroie.github.io/hasp-docs/#example-1/). I am not able to move between the pages with the navigation buttons located at the bottom of the pages. From Web UI screenshot menu it is possible to switch pages so command works.

How to trigger command to actually happen from an object on a page?

**Comment by fvanroie** [comment, marked as answer]
When you push a button, that button sends an MQTT event e.g. `hasp/plate/state/p1b1` => `DOWN`.

You need to capture this event in an Home Automation system like OpenHAB, NodeRed or Home Assistant and then send a `page` command in reply to this event.

**Reply by nagyrobi** [reply]
So hasp-lvgl can't handle this on its own? Need to create network traffic for pagination??

**Reply by fvanroie** [reply]
It was conceived as on online device, where all actions are published and received over MQTT.
The automations are handled off the device and the plate only processes commands from a remote controller.
Only recently with the addition of *groups*, there has been a way to control GPIOs locally.

A real *offline* mode is not available and is discussed [here](https://github.com/fvanroie/hasp-lvgl/discussions/61).

**Reply by nagyrobi** [reply]
I understand how it is.
However in real life this can cause troubles, and the topic you linked is important. Local controls like relays should to be available offline and page navigation too. Think of a WiFi outage due to a faulty AP firmware update. Suddenly you won't be able to turn on any lights in the house...

**Reply by fvanroie** [reply]
This has been added to 0.4.0. I will update the docs soon.

**Reply by nagyrobi** [reply]
Great!
